rpcs3/Emu/Cell/lv2/sys_fs.cpp

```diff
@@ -824,6 +824,12 @@ lv2_file::open_result_t lv2_file::open(std::string_view vpath, s32 flags, s32 mo
 		return {CELL_ENOTMOUNTED, path};
 	}
 
+	if (vpath == "/dev_hdd1")
```

Comment:
What if this happens not in CellOS process?

Reply:
In real PS3, with the file_system parameter of sys_fs_mount() set to "CELL_FS_SIMPLEFS", the device is mounted as a file (raw), instead of a directory.
But I'm not sure if we are capable of implementing the raw filesystem mounting in RPCS3, so I put the trick in sys_fs_open to redirect the file path to /dev_hdd1/caches/vsh/loop.

Reply:
I don't see any check for simplefs here

Reply:
The loop file is created by sys_fs_mount, which checks the filesystem parameter.

Reply:
It would be **more ideal** if we could actually emulate the mount process of CELL_FS_SIMPLEFS in sys_fs_mount() or vfs::mount(), rather than put a redirection trick in sys_fs_open() though.

Reply:
@Nekotekina @Megamouse How about the new implementation?

Reply:
It worked like a charm:
```
sys_fs: sys_fs_mount(dev_name=“CELL_FS_IOS:PATA1_HDD_DRIVE”, file_system=“CELL_FS_FAT”, path=“/dev_hdd1”, unk1=0x0, prot=0x1, unk3=0x0, str1=«NULL», str_len=0)
VFS: Mounted path "/dev_hdd1" to "RPCS3/dev_hdd1/caches/vsh/"
sys_fs: sys_fs_unmount(path=“/dev_hdd1”, unk1=0x0, unk2=0x1)
VFS: About to unmount '/dev_hdd1'
VFS: Unmounting 'dev_hdd1' = 'RPCS3/dev_hdd1/caches/vsh/'
sys_fs: sys_fs_mount(dev_name=“CELL_FS_UTILITY:HDD1”, file_system=“CELL_FS_SIMPLEFS”, path=“/dev_hdd1”, unk1=0x0, prot=0x0, unk3=0x0, str1=«NULL», str_len=0)
VFS: Mounted path "/dev_hdd1" to "RPCS3/dev_hdd1/caches/vsh/loop"
sys_fs: sys_fs_open(path=“/dev_hdd1”, flags=02, fd=*0x718330, mode=03000, arg=*0x0, size=0x0)
sys_fs: sys_fs_open(): fd=18, Regular file, “/dev_hdd1”, Mode: 0x600, Flags: 0x2, Pos/Size: 0/3.99999MB (0x0/0x3ffff8)
sys_fs: sys_fs_fget_block_size(fd=18, sector_size=*0x718338, block_size=*0x718340, arg4=*0x718348, out_flags=*0xd017ab10)
sys_fs: sys_fs_close(fd=18): Regular file, “/dev_hdd1”, Mode: 0x600, Flags: 0x2, Pos/Size: 1.5KB/3.99999MB (0x600/0x3ffff8)
sys_fs: sys_fs_unmount(path=“/dev_hdd1”, unk1=0x0, unk2=0x1)
VFS: About to unmount '/dev_hdd1'
VFS: Unmounting 'dev_hdd1' = 'RPCS3/dev_hdd1/caches/vsh/loop'
```